blocks/init/src/Blocks/custom/lottie/assets/lottie-control.js

```diff
@@ -0,0 +1,232 @@
+			this.lottieParams[breakpointName] = {
+				...lottieRespParams,
+				...this.lottieParams[breakpointName],
+			};
+			this.lottieContainerParams[breakpointName] = {
+				...containerRespParams,
+				...this.lottieContainerParams[breakpointName],
+			};
+
+			lottieRespParams = this.lottieParams[breakpointName];
+			containerRespParams = this.lottieContainerParams[breakpointName];
+		});
+	};
+
+	/**
+	 * Extract data from data attributes and set them to breakpoints.
+	 */
+	prepareLottieParams = () => {
+		const lottieContainers = this.lottieElement.querySelectorAll(this.lottieContainerSelector);
+
+		if (lottieContainers.length === 0) return;
+
+		const {
+			loop,
+			path,
+			viewportOffsetFactor,
+			elementOffsetFactor,
+			timelineOffsetFactor,
+		} = lottieContainers[0].dataset;
+
+		// Fallback values that will be inherited for each breakpoint.
+		let containerParamsFallback = {
+			viewportOffsetFactor,
+			elementOffsetFactor,
+			timelineOffsetFactor,
+		};
+
+		let lottieParamsFallback = {
+			loop,
+			path,
+		};
+
+		[...lottieContainers].forEach((lottieContainer) => {
+			const {
+				breakpoint,
+				loop,
+				path,
+				viewportOffsetFactor,
+				elementOffsetFactor,
+				timelineOffsetFactor,
+			} = lottieContainer.dataset;
+
+			if (!breakpoint) return;
+
+			// Container storage per breakpoint.
+			this.lottieContainers[breakpoint] = lottieContainer;
+
+			// Container parameters storage per breakpoint.
+			this.lottieContainerParams[breakpoint] = {
+				...containerParamsFallback,
+				...cleanObject({
+					viewportOffsetFactor,
+					elementOffsetFactor,
+					timelineOffsetFactor,
+				}),
+			};
+
+			// Lottie animation parameters storage per breakpoint.
+			this.lottieParams[breakpoint] = {
+				...lottieParamsFallback,
+				...cleanObject({
+					container: lottieContainer,
+					path,
+					renderer: 'svg',
+					progressiveLoad: true,
+					loop: loop === 'true',
+				}),
+			};
+
+			containerParamsFallback = this.lottieContainerParams[breakpoint];
+			lottieParamsFallback = this.lottieParams[breakpoint];
+		});
+
+		this.setParamsToAllBreakpoints();
+	};
+
+	/**
+	 * Detect the current breakpoint and return it.
+	 */
+	detectBreakpoint = () => {
+		let [breakpoint] = this.breakpoints[0];
+		this.breakpoints.forEach(([breakpointName, value]) => {
+			const mediaBreakpoint = window.matchMedia(`(max-width: ${value}px)`);
+			if (mediaBreakpoint.matches) {
+				breakpoint = breakpointName;
+			}
+		});
+
+		return breakpoint;
+	};
+
+	/**
+	 * Lottie animation control is being handled via custom events on each lottie container.
+	 *
+	 * activateBreakpointAnimation - Show and activate animation depending on the breakpoint.
+	 * deactivateBreakpointAnimation - Hide and deactivate animation depending on the breakpoint.
+	 */
+	activateBreakpointAnimation = () => {
+		this.currentLottie.container.classList.remove(this.IS_HIDDEN_CLASS);
+		const containerEvent = new Event('start');
```

Comment:
this is really generic event name please use something more descriptive of the usage